pkg/controller/sparkapplication/controller.go

```diff
@@ -401,8 +406,10 @@ func (c *Controller) getAndUpdateExecutorState(app *v1beta2.SparkApplication) er
 					c.recordExecutorEvent(app, newState, pod.Name)
 				}
 			}
-			executorStateMap[pod.Name] = newState
-
+			// If the executor number is higher than the `executorsProcessingLimit` we want to stop persisting executors
+			if executorID, _ := strconv.Atoi(getSparkExecutorID(pod)); executorID <= c.executorsProcessingLimit {
```

Comment:
since we don't save the executor in the map, we will hit the `!exists` condition above and keep emitting events for all executors with `ID > executorsProcessingLimit`

should we completely ignore executors that have `ID > executorsProcessingLimit` instead?

Reply:
I kind of like the idea of the kubernetes events to keep being reported. Do we see problems with that ?

Reply:
in that particular case, we send duplicated events for the same pod with the same state since this pod doesn't exist in the `app.Status.ExecutorState` list

```go
oldState, exists := app.Status.ExecutorState[pod.Name]
				// Only record an executor event if the executor state is new or it has changed.
				if !exists || newState != oldState {
					if newState == v1beta2.ExecutorFailedState {
```

for that edge case, we decided to not send the event as well, see my last commit.